### ISSUE TITLE
Should JsonWebToken.TryGetPayloadValue() support IDictionary<TKey, TValue>?

### ISSUE THREAD

**Issue by kevinchalet** [AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet#2281]
The following snippet - using the concrete `Dictionary<TKey, TValue>` now works fine on preview4:

```csharp
var handler = new JsonWebTokenHandler();
var token = handler.ReadJsonWebToken("eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhIjp7InByb3AxIjpbInZh" +
    "bHVlMSIsInZhbHVlMiJdfSwiZXhwIjoxNjkyNzA2ODAzLCJpYXQiOjE2OTI3MDMyMDMsIm5iZiI6MTY5MjcwMzIwM30.dZ" +
    "lSC8OLOKUvSCXu9uZFb91NQwWBJ7dwSxEn-WA2RVEBW1l3Wu2wdobOuezivtzX9OWpgYyvzog4nHfD-Yco4Ou5qYqN0JbB" +
    "fU7gESA95iDtDhbDzVB8T3W3T7k3a1LHwVNZbBGTnZQMgF447R-euIlMcGY6p5WQFLyFPZjBOKW2nlwh0MYQQ-5mZpEgzE" +
    "pdcHVOjdO0pV1EJTKFjHuiUuFsI6d6i3dLOQYX2JpaclT--pTFqs4Nx1O8gn9bKbfKTpGlpy2CpJJLmmzKqsLWtrXL0eSD" +
    "QQBK0ynvd6ssRARH_L3f9Hf05oBRqiqO3ukWw8qAoBE-ZF6906gYnwMcYA");

if (token.TryGetPayloadValue("a", out Dictionary<string, string[]> dictionary))
{
    Console.WriteLine("The dictionary has been correctly deserialized.");
}
```

This one, using `IDictionary<TKey, TValue>` as the generic type, doesn't:

```csharp
var handler = new JsonWebTokenHandler();
var token = handler.ReadJsonWebToken("eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhIjp7InByb3AxIjpbInZh" +
    "bHVlMSIsInZhbHVlMiJdfSwiZXhwIjoxNjkyNzA2ODAzLCJpYXQiOjE2OTI3MDMyMDMsIm5iZiI6MTY5MjcwMzIwM30.dZ" +
    "lSC8OLOKUvSCXu9uZFb91NQwWBJ7dwSxEn-WA2RVEBW1l3Wu2wdobOuezivtzX9OWpgYyvzog4nHfD-Yco4Ou5qYqN0JbB" +
    "fU7gESA95iDtDhbDzVB8T3W3T7k3a1LHwVNZbBGTnZQMgF447R-euIlMcGY6p5WQFLyFPZjBOKW2nlwh0MYQQ-5mZpEgzE" +
    "pdcHVOjdO0pV1EJTKFjHuiUuFsI6d6i3dLOQYX2JpaclT--pTFqs4Nx1O8gn9bKbfKTpGlpy2CpJJLmmzKqsLWtrXL0eSD" +
    "QQBK0ynvd6ssRARH_L3f9Hf05oBRqiqO3ukWw8qAoBE-ZF6906gYnwMcYA");

if (token.TryGetPayloadValue("a", out IDictionary<string, string[]> dictionary))
{
    Console.WriteLine("The dictionary has been correctly deserialized.");
}
```

Is it deliberate or an oversight? 😅 

/cc @jennyf19 @brentschmaltz

**Comment by brentschmaltz** [comment]
@kevinchalet we thank you so much for helping this effort.

The generic wasn't an oversight.
We explored a way of making that work, that will reduce the complexity by using IsAssignableFrom.
Which we use when we write objects [here](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/db6f21c0edda8588df33af010c8e69f3a9211b50/src/Microsoft.IdentityModel.Tokens/Json/JsonSerializerPrimitives.cs#L798).
We may have time for this.

Right now, we are focused on hardening and perf.

**Comment by jennyf19** [comment]
@kevinchalet We discussed and have decided to not fix this. Going to close, but feel free to reopen if you disagree.

**Comment by kevinchalet** [comment]
@jennyf19 that's fine by me. That said, it's not super clear what types are supported. May I suggest adding some type checks or updating the XML documentation to clarify what types exactly are supported?

**Comment by kevinchalet** [comment]
> @kevinchalet we thank you so much for helping this effort.

My pleasure. It's a win-win situation: all the bugs fixed before 7.0 ships are bugs I won't have to work around in OpenIddict 😄

It's nice to see Wilson is getting a lot more attention from Microsoft. It clearly needed more more love 😄
